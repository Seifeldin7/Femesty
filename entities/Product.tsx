export default class Product {
    public id!: number;
    public name: string;
    public description: string;
    public price: number;
    public sizes: string[];
    public colors: string[];
    public image1: any;
    public image2: any;
    public image3: any;
    public category: any;
    public seller: any;
    public created_at: Date;
  
    constructor(
      id: number,
      name: string,
      description: string,
      price: number,
      sizes: string[],
      colors: string[],
      image1: any,
      image2: any,
      image3: any,
      category: any,
      seller: any,
      created_at: Date
    ) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.price = price;
      this.image1 = image1;
      this.image2 = image2;
      this.image3 = image3;
      this.sizes = sizes;
      this.category = category;
      this.created_at = created_at;
      this.seller = seller;
      this.colors = colors;
    }
  }
  