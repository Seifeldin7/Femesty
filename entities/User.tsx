export default class User {
    public id!: string;
    public email: string;
    public username: string;
    public name: string;
    public image: string;
    public description: string;
    public phone: string;
    public address: string;
    public created_at: string;
    public password_reset_pin!: string;

    constructor(email: string,
        created_at: string,
        username: string,
        name: string,
        image: string,
        description: string,
        phone: string,
        address: string) {
        this.email = email;
        this.created_at = created_at;
        this.address = address;
        this.description = description;
        this.image = image;
        this.name = name;
        this.phone = phone;
        this.username = username;
    }
}
