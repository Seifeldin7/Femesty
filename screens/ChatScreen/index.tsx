import { Ionicons } from "@expo/vector-icons";
import React, { Component } from "react";
import { View, StatusBar, Text, TouchableOpacity } from "react-native";
import { FlatList } from "react-native-gesture-handler";
import { connect } from "react-redux";
import { ChatInputBox } from "../../components/ChatScreenComponents/ChatInputBox";
import { ChatMessage } from "../../components/ChatScreenComponents";
import { styles } from "./styles";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";

export class NCChatScreen extends Component<Props, State> {
  constructor(props: Readonly<Props>) {
    super(props);

    this.state = {};
    this.sendMessage = this.sendMessage.bind(this);
  }

  sendMessage = async (message: any) => {
    try {
      await this.props.AddMessage(message);
    } catch (err) {
      console.log(err);
    }
  };

  async componentDidMount() {
    try {
      await this.props.FetchMessages();
    } catch (err) {
      console.log(err);
    }
  }

  render() {
    return (
      <ScreenSkeleton
        goBackNavigation={() => this.props.navigation.goBack()}
        title={"Live Chat"}
      >
        <FlatList
          data={this.props.messages.messages}
          renderItem={({ item }) => <ChatMessage message={item} />}
          keyExtractor={(item: any) => item.id.toString()}
          inverted
        />
        <ChatInputBox
          sendMessageFunction={(message: any) => this.sendMessage(message)}
        />
      </ScreenSkeleton>
    );
  }
}

export const ChatScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(NCChatScreen);
