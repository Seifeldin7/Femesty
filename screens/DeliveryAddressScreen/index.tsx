import React, { Component } from "react";
import { View } from "react-native";
import { styles } from "./styles";
import { connect } from "react-redux";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { Form } from "../../components/SharedComponents/Form";
import { Button } from "../../components/SharedComponents/Buttons";
import { BasicColors, MainColors } from "../../styles/Colors";
import { ButtonStyles } from "../../styles/Buttons";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";
import { DoubleVz21Container } from "../../styles/Containers";

export class NCDeliveryAddressScreen extends Component<Props, State>
{
    constructor(props: Readonly<Props>) {
        super(props);
        this.state = {
            submitIsPressed: false,
            formValues: {}
        };
        this.onConfirm = this.onConfirm.bind(this);
        this.updateAddress = this.updateAddress.bind(this);
    }
    async onConfirm(formValues: any) {
        await this.props.SENDCODE(formValues.phone);
        this.setState({ formValues: formValues });
        this.props.navigation.navigate("PhoneConfirmation", {
            verify: this.props.VERIFYCODE,
            updateAddress: this.updateAddress, 
            cancelcode: this.props.CANCELCODE
        });
    }
    async updateAddress() {
        let userAddress = this.state.formValues;
        this.props.EDITADDRESS(userAddress.name, userAddress.phone, userAddress.address);
        this.props.navigation.navigate("Profile");
    }
    render() {
        return (
            <ScreenSkeleton
                goBackNavigation={() => { this.props.navigation.navigate("Profile") }}
                title={"Delivery Details"}
                rightControlButtonNavigation={() =>
                    this.props.navigation.navigate("Profile")
                }
            >
                <View style={DoubleVz21Container.Container}>
                    <View style={{ ...DoubleVz21Container.TopComponent, ...styles.row2 }}>
                        <Form
                            submitIsPressed={this.state.submitIsPressed}
                            onSubmit={this.onConfirm}
                            formData={{
                                name: this.props.profile.profile.name,
                                phone: this.props.profile.profile.phone,
                                address: this.props.profile.profile.address

                            }}
                            inputProps={{
                                name: {
                                    label: "Name",
                                    placeholder: "Jane Doe",
                                },
                                phone: {
                                    label: "Phone Number",
                                    placeholder: "+20213456789",
                                    autoCompleteType: "tel",
                                    errorText: "Please enter a valid phone number!"

                                },
                                address: {
                                    label: "Address",
                                    placeholder: "13th street Egypt square",
                                }
                            }} />
                    </View>
                    <View style={{ ...DoubleVz21Container.BottomComponent, ...styles.row3 }}>
                        <Button
                            style={{ ...ButtonStyles.Large, alignSelf: 'center' }}
                            textColor={BasicColors.White}
                            backgroundColor={MainColors.Light}
                            enable={true}
                            title="Confirm"
                            onPress={() => this.setState({ submitIsPressed: true })} />
                    </View>
                </View>
            </ScreenSkeleton>
        );
    }
}

export const DeliveryAddressScreen = connect(mapStateToProps, mapDispatchToProps)(NCDeliveryAddressScreen);
