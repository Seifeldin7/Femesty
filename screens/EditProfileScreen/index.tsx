import React, { Component } from "react";
import {
  View,
  Text,
  TouchableOpacity,
} from "react-native";
import { connect } from "react-redux";
import { styles } from "./styles";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { Form } from "../../components/SharedComponents/Form";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";
import { TripleVz123Container } from "../../styles/Containers";
import { Avatar } from "../../components/ThemeComponents/Avatar";

export class NCEditProfileScreen extends Component<Props, State> {
  constructor(props: Readonly<Props>) {
    super(props);

    this.state = {
      submitIsPressed: false
    };
    this.saveChanges = this.saveChanges.bind(this);
  }

  async saveChanges(formValues: any) {
    await this.props.EditProfile(formValues.name, formValues.username);
    this.props.navigation.goBack();
  }

  render() {
    return (
      <ScreenSkeleton
        goBackNavigation={() => this.props.navigation.goBack()}
        title={"Edit Profile"}
      >
        <View style={TripleVz123Container.Container}>
          <View style={TripleVz123Container.TopComponent}>
            <TouchableOpacity
              style={styles.saveView}
              onPress={() => {
                this.setState({ submitIsPressed: true })
                this.props.navigation.goBack();
              }}
            >
              <Text style={styles.save}>Save Changes</Text>
            </TouchableOpacity>
          </View>
          <View style={[TripleVz123Container.MiddleComponent, { alignSelf: 'center' }]}>
            {this.props.profile.profile.image ? (
              <Avatar imgPath={{ uri: this.props.profile.profile.image }}
                size="large" borderRadius="rounded-rectangle" />
            ) : (
              <Avatar imgPath={require("../../assets/images/default_profile_pic.jpg")}
                size="large" borderRadius="rounded-rectangle" />
            )}
          </View>
          <View style={[TripleVz123Container.BottomComponent, styles.inputs]}>
            <Form
              submitIsPressed={this.state.submitIsPressed}
              onSubmit={this.saveChanges}
              formData={{
                name: this.props.profile.profile.name,
                username: this.props.profile.profile.username,
              }}
              inputProps={{
                name: {
                  label: "Name",
                  placeholder: "@name",
                },
                username: {
                  label: "User Name",
                  placeholder: "@username",

                },
              }} />
          </View>
        </View>
      </ScreenSkeleton>
    );
  }
}

export const EditProfileScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(NCEditProfileScreen);
