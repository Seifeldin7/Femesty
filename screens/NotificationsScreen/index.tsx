import React, { Component } from "react";
import {
  View,
  StatusBar,
  Text,
  RefreshControl,
  ActivityIndicator,
  TouchableOpacity,
} from "react-native";
import { styles } from "./styles";
import { connect } from "react-redux";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { SwipeListView } from "react-native-swipe-list-view";
import { VisibleItem, HiddenItem } from "../../components/NotificationsScreenComponents";
import { MainColors } from "../../styles/Colors";
import { NotificationScreenLoading } from "../../components/ThemeComponents/Views/LoadingViews";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";

export class NCNotificationScreen extends Component<Props, State> {
  constructor(props: Readonly<Props>) {
    super(props);

    this.state = {
      notifications: [],
      refreshing: false,
      loadMore: false,
      hasScrolled: false,
    };
    this.markAsRead = this.markAsRead.bind(this);
    this.markAllAsRead = this.markAllAsRead.bind(this);
    this.onLoadMore = this.onLoadMore.bind(this);
    this.renderItem = this.renderItem.bind(this);
    this.renderLoader = this.renderLoader.bind(this);
    this.onRefresh = this.onRefresh.bind(this);
    this.renderHiddenItem = this.renderHiddenItem.bind(this);
  }

  async markAsRead(id: number) {
    await this.props.MarkAsRead(id);
  }

  async markAllAsRead() {
    await this.props.MarkAllAsRead();
  }

  async onLoadMore() {
    if (!this.state.hasScrolled) {
      return null;
    }
    this.setState({ loadMore: true });
    await this.props.FetchNotifications();
    this.setState({ loadMore: false });
  }

  renderItem = (data: any, rowMap: any) => {
    return <VisibleItem data={data} />;
  };

  renderHiddenItem = (data: any) => {
    return <HiddenItem data={data} onMarkAsRead={this.markAsRead} />;
  };

  renderLoader = () => {
    return this.state.loadMore ? (
      <View style={styles.loader}>
        <ActivityIndicator
          size="large"
          style={styles.loader}
          color={MainColors.Dark}
        />
      </View>
    ) : null;
  };

  async componentDidMount() {
    this.setState({ refreshing: true });
    await this.props.FetchNotifications();
    this.setState({ refreshing: false });
  }

  async onRefresh() {
    this.setState({ refreshing: true });
    await this.props.RefreshNotifications();
    await this.props.FetchNotifications();
    this.setState({ refreshing: false });
  }

  onScroll = () => {
    this.setState({ hasScrolled: true });
  };

  render() {
    return (
      <ScreenSkeleton
        goBackNavigation={() => this.props.navigation.goBack()}
        title={"My Notifications"}
      >
        <TouchableOpacity
          style={styles.saveView}
          onPress={() => this.markAllAsRead()}
        >
          <Text style={styles.save}>Mark all as read</Text>
        </TouchableOpacity>
        {this.state.refreshing ? (
            <NotificationScreenLoading />
        ) : (
          <View style={{ flex: 5.2 }}>
            <SwipeListView
              onScroll={this.onScroll}
              keyExtractor={(item: any) => item.id.toString()}
              scrollEnabled={true}
              style={{ marginTop: 20 }}
              data={this.props.notifications.notifications}
              renderItem={this.renderItem}
              renderHiddenItem={this.renderHiddenItem}
              rightOpenValue={-120}
              disableRightSwipe
              leftActivationValue={100}
              rightActivationValue={-200}
              leftActionValue={0}
              rightActionValue={0}
              onEndReachedThreshold={0.01}
              onEndReached={() => {
                this.onLoadMore();
                this.renderLoader();
              }}
              ListFooterComponent={this.renderLoader}
              refreshControl={
                <RefreshControl
                  refreshing={this.state.refreshing}
                  onRefresh={this.onRefresh}
                />
              }
            />
          </View>
        )}

      </ScreenSkeleton>
    );
  }
}

export const NotificationScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(NCNotificationScreen);
