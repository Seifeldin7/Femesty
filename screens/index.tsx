export * from "./ChatScreen"
export * from "./DeliveryAddressScreen"
export * from "./EditProfileScreen"
export * from "./FavoriteScreen"
export * from './HomeScreen';
export * from "./LoginScreen";
export * from "./NotificationsScreen"
export * from "./OrderScreen";
export * from "./PhoneConfirmationScreen";
export * from "./ProfileScreen"
export * from "./SearchResultScreen"
export * from "./SearchScreen"
export * from "./SplashScreen"