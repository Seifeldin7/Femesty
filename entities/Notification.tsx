import Order from "./Order";
import Product from "./Product";
import User from "./User";

export default class Notification {
    public id: number;
    public name: string;
    public description: string;
    public created_at: Date;
    public is_read: boolean;
    public subject: Product | Order;
    public subject_type: string;
    public user: User;
    constructor(
        id: number,
        name: string,
        description: string,
        created_at: Date,
        is_read: boolean,
        subject: Product | Order,
        subject_type: string,
        user: User,
    ) {
        this.id = id,
        this.name = name;
        this.description = description;
        this.created_at = created_at;
        this.is_read = is_read;
        this.subject = subject;
        this.subject_type = subject_type;
        this.user = user;
    }
}
