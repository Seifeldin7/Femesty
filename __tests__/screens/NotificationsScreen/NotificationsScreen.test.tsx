import React from "react";
import { fireEvent, render } from "react-native-testing-library";
import { ReactTestInstance } from "react-test-renderer";
import Notification from "../../../entities/Notification";
import Product from "../../../entities/Product";
import User from "../../../entities/User";
import { NCNotificationScreen } from "../../../screens/NotificationsScreen";


jest.mock('../../../services/ApiClient', () => ({
    constants: {
        manifest: {
            developer: {
                projectRoot: '/home/test/project',
            },
            logUrl: 'https://localhost:19001/logs',
            extra: {
                baseUrl: '192.168.1.1'
            }
        }
    },
}));

test('should render notifications correctly', () => {
    const not = new Notification(
        1,
        "not1",
        "Item is shipped",
        new Date,
        false,
        new Product(1,
            'JEANS',
            'Blue Jeans',
            20,
            ['XL', 'S'],
            ['blue'],
            '',
            '',
            '',
            'pants',
            { name: 'ZARA' },
            new Date), "Product", new User("s.seif@yahoo.com",
                "11 Nov 2020",
                "",
                "",
                "",
                "Famous Designer",
                "01111111",
                "Mokattam"));
    const props = {
        FetchNotifications: jest.fn(),
        MarkAsRead: jest.fn(),
        MarkAllAsRead: jest.fn(),
        RefreshNotifications: jest.fn(),
        navigation: { goBack: jest.fn(), reset: jest.fn(), navigate: jest.fn() },
        notifications: {
            notifications: [not],
        }
    };

    const tree = render(
        <NCNotificationScreen
            {...props}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();

});

describe('<Notifications>', () => {
    let getByText: (text: string) => ReactTestInstance | null;
    let getByTestId: (testID: string) => ReactTestInstance | null;
    const fetchNotifications = jest.fn();
    const markAllAsRead = jest.fn();
    const notifications = [new Notification(
        1,
        "not1",
        "Item is shipped",
        new Date,
        false,
        new Product(1,
            'JEANS',
            'Blue Jeans',
            20,
            ['XL', 'S'],
            ['blue'],
            '',
            '',
            '',
            'pants',
            { name: 'ZARA' },
            new Date), "Product", new User("s.seif@yahoo.com",
                "11 Nov 2020",
                "",
                "",
                "",
                "Famous Designer",
                "01111111",
                "Mokattam")),
    new Notification(
        2,
        "not2",
        "Item is returned",
        new Date,
        false,
        new Product(1,
            'JEANS',
            'Blue Jeans',
            20,
            ['XL', 'S'],
            ['blue'],
            '',
            '',
            '',
            'pants',
            { name: 'ZARA' },
            new Date), "Product", new User("s.seif@yahoo.com",
                "11 Nov 2020",
                "",
                "",
                "",
                "Famous Designer",
                "01111111",
                "Mokattam"))];
    const props = {
        FetchNotifications: fetchNotifications,
        MarkAsRead: jest.fn(),
        MarkAllAsRead: markAllAsRead,
        RefreshNotifications: jest.fn(),
        navigation: { goBack: jest.fn(), reset: jest.fn(), navigate: jest.fn() },
        notifications: {
            notifications: notifications,
        }
    };
    beforeEach(() => {
        ({ getByText, getByTestId } = render(
            <NCNotificationScreen
                {...props}
            />
        ))
    });
    test('fetch function to be called', () => {
        expect(fetchNotifications).toHaveBeenCalledTimes(1);
    });
    test('mark all as read', () => {
        const markAllAsReadBtn = getByText("Mark all as read");
        expect(markAllAsReadBtn).toBeDefined();
        if(markAllAsReadBtn) {
            fireEvent.press(markAllAsReadBtn);
            expect(markAllAsRead).toHaveBeenCalledTimes(1);
        }
    });
});
