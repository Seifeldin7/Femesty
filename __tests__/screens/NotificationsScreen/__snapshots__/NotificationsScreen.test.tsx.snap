// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render notifications correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "marginTop": null,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": -20,
          "opacity": 1,
          "paddingTop": 7,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "marginLeft": 20,
          "width": 170,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        Notifications
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 40,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "marginHorizontal": 10,
        "marginTop": 30,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1E1E1E",
          "fontFamily": "Lato_SemiBold",
          "fontSize": 15,
        }
      }
    >
      Mark all as read
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1334,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="black"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
